doc/proposals/helm-operator.md

```diff
@@ -0,0 +1,127 @@
+  * Package functions:
+    * NewManager - method that returns a new Manager for a provided helm chart.
+    * NewManagersFromEnv - method that returns a map of GVK to Manager types based on environment variables.
+    * NewManagersFromFile - method that returns a map of GVK to Manager types based on a provided config file.
+
+### Commands
+
+We are adding and updating existing commands to accommodate the helm operator.  Changes to the `cmd` package as well as changes to the generator are needed.
+
+#### New
+
+New functionality will be updates to allow Helm operator developers to create a new boilerplate operator structure with everything necessary to get started developing and deploying a Helm operator with the SDK.
+
+```
+operator-sdk new <project-name> --type=helm --kind=<kind> --api-version=<group/version>
+```
+
+This will be new scaffolding for the above command under the hood. We will:
+* Create a `./<project-name>` directory.
+* Create a `./<project-name>/charts` directory.
+* Generate a simple default chart at `./<project-name>/charts/<kind>`.
+* Create a new watches file at `./<project-name>/watches.yaml`. The chart and GVK will be defaulted based on input to the `new` command.
+* Create a `./<project-name>/deploy` with the Kubernetes resource files necessary to run the operator.
+* Create a `./build/Dockerfile` that uses the watches file and the helm chart. It will use the Helm operator as its base image.
+
+The resulting structure will be:
+
+```
+<project-name>
+|   watches.yaml
+|
+|-- build
+|   |   Dockerfile
+|
+|-- charts
+|   |-- <kind>
+|       |   Chart.yaml
+|       |   ...
+|
+|-- deploy
+|   |   operator.yaml
+|   |   role_binding.yaml
+|   |   role.yaml
+|   |   service_account.yaml
+|   |
+|   |-- crds
+|       |   <gvk>_crd.yaml
+|       |   <gvk>_cr.yaml
+```
+
+#### Add
+
+Add functionality will be updated to allow Helm operator developers to add new CRDs/CRs and to update the watches.yaml file for additional Helm charts. The command helps when a user wants to watch more than one CRD for their operator.
+
+```
+operator-sdk add api --api-version=<group>/<version> --kind=<kind>
+```
+
+Flags:
+* **Required:** --kind - the kind for the CRD.
+* **Required:** --api-version - the group/version for the CRD.
+
+**NOTE:** `operator-sdk add controller` will not be supported, since it doesn't make sense for a Helm operator.
+
+#### Up
+
+Up functionality will be updated to allow Helm operator developers to run their operator locally, using the `operator-sdk` binary's built-in helm operator implementation.
+
+```
+operator-sdk up local
+```
+
+This should use the known structure and the helm operator code to run the operator from this location. The existing code will need to be updated with a new operator type check for `helm` (in addition to existing `go` and `ansible` types). The command works by running the operator-sdk binary, which includes the Helm operator code, as the operator process.
```

Comment:
Maybe this needs to be solved later, and that is fine, but how are we differentiating between ansible and helm?

Reply:
For helm at least, the immediately obvious option would be to check for the presence of the `helm-charts` directory in the project root.